src/Misc/layoutbin/daemon.svc.sh.template

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+SVC_NAME=`echo -n "{{SvcNameVar}}" | tr -c "[:alnum:]" "_"`
+SVC_DESCRIPTION="{{SvcDescription}}"
+
+SVC_CMD=$1
+arg_2=${2}
+
+AGENT_ROOT=`pwd`
+
+UNIT_PATH=/etc/rc.d/${SVC_NAME}
+TEMPLATE_PATH=./bin/vsts.agent.daemon.template
+TEMP_PATH=./bin/vsts.agent.daemon.temp
+CONFIG_PATH=.service
+
+user_id=`id -u`
+
+# daemon must run as sudo
+# this script is a convenience wrapper around daemon
+if [ $user_id -ne 0 ]; then
+    echo "Must run as sudo"
```

Comment:
```suggestion
    echo "Must run as root"
```

FreeBSD does not come with `sudo` or `doas` out of the box and the recent `sudo` vulnerabilities do not make me want to recommend that people install it.

Reply:
What would be your alternative to `sudo`?
I tried the following command, but it failed because it could not determine the "original" user with `$SUDO_USER`:
```
guest@FreeBSD:~/agent01 % su root -c './svc.sh install'
Password:
Creating launch agent in /etc/rc.d/vsts_agent_azdevops_MyPool_FreeBSD_service
Run as user: 
Run as uid: 0
gid: 0
su: unknown class: env > "/home/guest/agent01/.env_file"
Failed: failed to create environment file
```
Are there other commands than `su` (and of course `sudo`) that work with `$SUDO_USER`, or is there a better way to determine the "original" user in `svc.sh`?

Reply:
Depending on `SUDO_USER` makes me really nervous.  If this is ever run by root in a situation where the environment is not properly sanitised, we have an exciting security vulnerability.

What is the purpose of this script?  It looks like it's an rc script, but it's not using any of the normal rc machinery.

Reply:
This was originally taken from Linux path

https://github.com/microsoft/azure-pipelines-agent/blob/7db4d8367d03b5dec00bf307cd4bbc40f6e58454/src/Misc/layoutbin/systemd.svc.sh.template#L53-L55


While this could/should be improved, this behavior exist in shipping product.

Reply:
OK, so I think we should keep the current behavior in this PR for now, but also keep in mind to change it for Linux and FreeBSD in a separate PR later. Do you agree with this?

Reply:
I've added an error message if `run_as_user` cannot be determined via `$SUDO_USER`. (see lines 56-58)
In this case you'll have to explicitly specify the username, but you don't need to install `sudo`.